src/devices/bus/sms_3d/s3dport.h

```diff
@@ -0,0 +1,87 @@
+// license:BSD-3-Clause
+// copyright-holders:Enik Land
+/**********************************************************************
+
+    Input Port for the Sega 3-D Glasses / SegaScope 3-D Glasses
+
+**********************************************************************
+
+
+**********************************************************************/
+
+#ifndef MAME_BUS_SMS_3D_PORT_H
+#define MAME_BUS_SMS_3D_PORT_H
+
+#include "screen.h"
+
+#pragma once
+
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> sms_3d_port_device
+
+class device_sms_3d_port_interface;
+
+class sms_3d_port_device : public device_t,
+								public device_slot_interface
```

Comment:
Why not use `device_single_card_slot_interface` to get more functionality with less boilerplate?

Reply:
Done.